collect_app/src/main/java/org/odk/collect/android/fragments/ShowQRCodeFragment.java

```diff
@@ -149,7 +149,12 @@ public void onClick(View v) {
             case R.id.btnSelect:
                 Intent photoPickerIntent = new Intent(Intent.ACTION_PICK);
                 photoPickerIntent.setType("image/*");
-                startActivityForResult(photoPickerIntent, SELECT_PHOTO);
+                if (photoPickerIntent.resolveActivity(getActivity().getPackageManager()) != null) {
+                    startActivityForResult(photoPickerIntent, SELECT_PHOTO);
+                } else {
+                    ToastUtils.showShortToast(getString(R.string.activity_not_found));
+                    Timber.w(getString(R.string.activity_not_found));
```

Comment:
But you should use as it is in this example `choose image` argument as well https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/widgets/ImageWidget.java#L213